src/dl_node.cpp

```diff
@@ -30,135 +31,43 @@ namespace ovms {
-            SPDLOG_DEBUG("Completion callback received for node name: {}", this->getName());
-            // After inference is completed, input blobs are not needed anymore
-            this->inputBlobs.clear();
-            notifyEndQueue.push(*this);
-            infer_request.SetCompletionCallback([]() {});  // reset callback on infer request
-        });
-        SPDLOG_DEBUG("Starting infer async for node name: {}", getName());
-        infer_request.StartAsync();
-    } catch (const InferenceEngine::details::InferenceEngineException& e) {
-        SPDLOG_DEBUG("[Node: {}] Exception occured when starting async inference or setting completion callback on model: {}, error: {}",
-            getName(), modelName, e.what());
-        return StatusCode::OV_INTERNAL_INFERENCE_ERROR;
-    } catch (const std::exception& e) {
-        SPDLOG_DEBUG("[Node: {}] Exception occured when starting async inference or setting completion callback on  model: {}, error: {}",
-            getName(), modelName, e.what());
-        return StatusCode::OV_INTERNAL_INFERENCE_ERROR;
-    } catch (...) {
-        SPDLOG_DEBUG("[Node: {}] Unknown exception occured when starting async inference or setting completion callback on model: {}",
-            getName(), modelName);
-        return StatusCode::OV_INTERNAL_INFERENCE_ERROR;
-    }
-    return StatusCode::OK;
-}
-
-Status DLNode::fetchResults(BlobMap& outputs) {
-    // ::execute needs to be executed before ::fetchResults
-    if (this->model == nullptr) {
-        SPDLOG_DEBUG("[Node: {}] Fetching results failed due to earlier execution failure", getName());
-        return StatusCode::UNKNOWN_ERROR;
-    }
-
-    // Get infer request corresponding to this node model
-    auto streamId = this->nodeStreamIdGuard->tryGetId();
-    if (!streamId) {
-        SPDLOG_DEBUG("[Node: {}] Fetching results failed - node had stream Id never assigned", getName());
-        return StatusCode::UNKNOWN_ERROR;
-    }
-    auto& infer_request = this->model->getInferRequestsQueue().getInferRequest(streamId.value());
+Status DLNode::fetchResults(BlobMap& outputs, InferenceEngine::InferRequest& inferRequest, ModelInstance& model, session_key_t sessionKey) {
     // Wait for blob results
-    SPDLOG_DEBUG("[Node: {}] Waiting for infer request with streamId: {} to finish", getName(), streamId.value());
-    auto ov_status = infer_request.Wait(InferenceEngine::IInferRequest::RESULT_READY);
-    SPDLOG_DEBUG("[Node: {}] Infer request with streamId: {} finished", getName(), streamId.value());
-    this->inputBlobs.clear();
+    SPDLOG_LOGGER_DEBUG(dag_executor_logger, "Node: {} session: {} Waiting for infer request to finish", getName(), sessionKey);
+    auto ov_status = inferRequest.Wait(InferenceEngine::IInferRequest::RESULT_READY);
+    SPDLOG_LOGGER_DEBUG(dag_executor_logger, "Node: {} session: {} infer request finished", getName(), sessionKey);
+
+    dynamic_cast<DLNodeSession&>(this->getNodeSession(sessionKey)).clearInputs();
```

Comment:
If we have separate session per one node inference, why do we need to clear inputs?

Reply:
Because we do not want to keep the inputs longer than necessary. That was the purpose for having clearInputs in dl_node.cpp as well.